client/network/src/protocol/light_client_handler.rs

```diff
@@ -1457,10 +1531,12 @@ mod tests {
 		let pset = peerset();
 		let mut behaviour = make_behaviour(true, pset.1, make_config());
 
-		behaviour.inject_connected(peer.clone(), empty_dialer());
+		behaviour.inject_connection_established(&peer, &ConnectionId::new(1), &empty_dialer());
+		behaviour.inject_connected(&peer);
```

Comment:
Should these two lines not be the other way around?

Reply:
I think this is right, see the [Swarm](https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/lib.rs#L542-L545) and [the docs](https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/behaviour.rs#L79-L80) (though the docs should read "when the _first_ connection to the peer is established ...", there is a typo there). It is the same order for `inject_connection_closed` / `inject_disconnected` as well as `inject_addr_reach_failure` / `inject_dial_failure`, i.e. the more granular callback is invoked first, followed by a conditional invocation of the second.

Reply:
Strictly speaking, for these tests, the calls to `inject_connected` are not even necessary since the impl of the behaviour is empty, I just added them for completeness and to not make the tests rely on such an empty impl but only on the contract `libp2p-swarm` gives for these callbacks.